File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys

input = sys.stdin.readline

n, m = map(int, input().split())

never_heard = []
result = []

for _ in range(n):
    name = input().rstrip()
    never_heard.append(name)

d = dict.fromkeys(never_heard, 0)

for _ in range(m):
    name = input().rstrip()
    if name in d:
        result.append(name)

result.sort()
print(len(result))
for k in result:
    print(k)

File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws Exception {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		
		HashSet<String> list = new HashSet<>();
		ArrayList<String> result = new ArrayList<>();
		
		for (int i = 0; i < n; i++) {
			list.add(br.readLine());
		}
		
		for (int j = 0; j < m; j++) {
			String name = br.readLine();
			if (list.contains(name)) {
				result.add(name);
			}
		}
		
		Collections.sort(result);
		sb.append(result.size() + "\n");
		for (String s : result) {
			sb.append(s + "\n");
		}
		
		System.out.print(sb);
	}

}
